feat: Promote FormField from Preview to Main (#2934)
Workday/canvas-kit: modules/docs/mdx/12.0-UPGRADE-GUIDE.mdx

In v12, all the components listed in this guide have started using our new
[Canvas Tokens Web](https://workday.github.io/canvas-tokens/?path=/docs/docs-getting-started--docs).

> You must add `@workday/canvas-tokens-web` to ensure our components are properly styled and the

Reviewer: Did we want to bold this like above? 
```suggestion
**You must** add `@workday/canvas-tokens-web` to ensure our components are properly styled and the
```

  labelPosition={FormField.LabelPosition.Left}
  useFieldSet={true}
  required={true}
  hintId="hint-alert"
  hintText="Please enter a valid email."
  label="Email"
>
  <TextInput onChange={handleChange} value={value} />
</FormField>

// v12 Newly promoted FormField from Preview to Main
<FormField
  error="error"
  orientation="horizontalStart"
  isRequired={true}
>
	<FormField.Label>Email</FormField.Label>
	<FormField.Field>
		<FormField.Input as={TextInput} onChange={handleChange} value={value} />
		<FormField.Hint>Please enter a valid email.</FormField.Hint>
	</FormField.Field>
</FormField>
```

**Noticeable changes:**

- `error` prop takes in the following values: `"error" | "alert"`.
- `labelPosition` becomes `orientation` with the following values:
  `"horizontal" | "horizontalStart" | "horizontalEnd" | "vertical"`.
- `useFieldSet` is no longer valid. If you want to group inputs, use
  [`FormFieldGroup`](#form-field-group) component.
- `required` becomes `isRequired`.
- `hintId` is no longer needed. The component handles associating the hint text, label and input
  automatically. If you wish to have a unique `id`, you can add one onto the `FormField` element.

> - `hintText` is no longer a valid prop. Use `FormField.Hint` sub component.

Reviewer: is "sub component" hyphenated? I'm not sure. 

```suggestion
- `hintText` is no longer a valid prop. Use `FormField.Hint` sub-component.
```

- `errorLabel` and `alertLabel` are no longer valid props. Customize your hint text inside of
  `FormField.Hint`.
- `label` is no longer a valid prop. Use `FormField.Label` sub component to render your label text.
- Instead of rendering an input, ensure you use `FormField.Input` with the `as` prop. This ensures
  proper error handling and aria attributes for accessibility.

🤖 The codemod will handle the change of `orientation="horizontal"` to
`orientation="horizontalStart"` if you're using the string literal values. It will also handle
updating your imports from `@workday/canvas-kit-preview-react/form-field` to
`@workday/canvas-kit-react/form-field`.

#### Form Field Group

> We've added a new component to the `form-field` package to allow users to group inputs without

Reviewer: Did you want this hyphenated or like the above? 
```suggestion
We've added a new component to the `FormField` package to allow users to group inputs without
```
